### PR TITLE
[AND-3604] MoPub native adapter

### PR DESCRIPTION
**Pull request by HuFangshuai**
Native ads adapter implementation.
Test app PR: https://github.com/Vungle/mopub-adapter-testapp-android/pull/52

(No e2e tests due to PROD env not getting deployed)

### CODE REVIEW

File: Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import static com.mopub.common.logging.MoPubLog.AdapterLogEvent.CUSTOM;
+import static com.mopub.common.logging.MoPubLog.AdapterLogEvent.SHOW_ATTEMPTED;
+
+public class VungleAdRenderer implements MoPubAdRenderer<VungleNative.VungleNativeAd> {
+
+    private final VungleViewBinder mViewBinder;
```
Comment (anton-blashchuk) [inline Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java]: subjective, but do we need "m" prefix?

File: Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+    /**
+     * A weak hash map used to keep track of view holder so that the views can be properly recycled.
+     */
+    private final WeakHashMap<View, VungleNativeViewHolder> mViewHolderMap;
```
Comment (anton-blashchuk) [inline Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java]: subjective, but do we need "m" prefix?

File: Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                            nativeAdLayout);
+            ViewGroup.LayoutParams layoutParams = adOptionsView.getLayoutParams();
+            if (layoutParams instanceof RelativeLayout.LayoutParams) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                    ((RelativeLayout.LayoutParams) layoutParams).addRule(
+                            RelativeLayout.ALIGN_PARENT_END);
+                } else {
+                    ((RelativeLayout.LayoutParams) layoutParams).addRule(
+                            RelativeLayout.ALIGN_PARENT_RIGHT);
+                }
```
Comment (anton-blashchuk) [inline Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java]: I think we're always >=JELLY_BEAN_MR_1

File: Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+        @Nullable
+        private TextView titleView;
+        @Nullable
+        private TextView textView;
+        @Nullable
+        private TextView callToActionView;
+        @Nullable
+        private ImageView adIconView;
+        @Nullable
+        private TextView mStarRatingTextView;
```
Comment (anton-blashchuk) [inline Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java]: subjective, but do we need "m" prefix? especially considering all  other member fields don't have it

**Review by anton-blashchuk — APPROVED** [review]
lgtm, only questions to naming style, not sure if we use "m" and "s" for prefixes of member and static fields respectively

File: Vungle/src/main/java/com/mopub/nativeads/VungleNative.java [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+                return;
+            }
+            mExtras.put(key, value);
+        }
+
+        /**
+         * Returns the String corresponding to the ad's title.
+         */
+        final public String getTitle() {
+            return nativeAd.getAdHeadline();
```
Comment (HuFangshuai) [inline Vungle/src/main/java/com/mopub/nativeads/VungleNative.java]: Here's the confusing API. @felixzhang0703 @ashinagawa

Comment (felixzhang0703) [reply Vungle/src/main/java/com/mopub/nativeads/VungleNative.java]: I suggest that we rename this api to nativeAd.getTitle() from sdk side. And in integration document, we can add description to this API. 
cc @HuFangshuai @ashinagawa

**Comment by jeffreychang-vungle** [comment]
Looks good, although need a more permanent branch

**Pull request merged**